scripts/formatters/apheleia-npx

```diff
@@ -36,30 +36,32 @@ find_upwards() {
 dir="$(find_upwards package.json)"
 
 if [[ -d $dir ]]; then
-    cd $dir
+    cd "$dir"
 
-    pnp_root=$(find_upwards '.pnp.cjs')
-    npm_root=$(find_upwards 'node_modules')
+    pnp_root="$(find_upwards .pnp.cjs)"
+    npm_root="$(find_upwards node_modules)"
 
     if [[ -n ${pnp_root} && ${#pnp_root} -gt ${#npm_root} ]]; then
         # trying pnp
         pnp_path="${pnp_root}/.pnp.cjs"
-        bin="$(${pnp_bin} ${pnp_path} $1)"
+        bin="$(${pnp_bin} "${pnp_path}" "$1")"
         # note: $bin might not be on the real filesystem,
         #   might be in a zip archive
         if [[ -n $bin ]]; then
+            node="$(which node)"
             if [[ -f "${pnp_root}/.pnp.loader.mjs" ]]; then
-                loader_opt="--loader ${pnp_root}/.pnp.loader.mjs"
+                exec ${node} --require "${pnp_path}" \
+                    --loader "${pnp_root}/.pnp.loader.mjs" "${bin}" "${@:2}"
             fi
-            node=$(which node)
-            command="${node} --require ${pnp_path} ${loader_opt} ${bin} ${@:2}"
-            exec ${command}
+            exec ${node} --require "${pnp_path}" "${bin}" "${@:2}"
```

Comment:
Are we doing two `exec`s now when .pnp.loader.mjs is present?

Reply:
Nah, `exec` is a terminal command. It completely replaces the current process with that of the invoked command - so, nothing after it in the script will execute. https://en.wikipedia.org/wiki/Exec_(system_call)